nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,543 @@
+    "EMAIL_CONFIG=${cfg.emailConfig}"
+    "DEFAULT_FROM_EMAIL=${cfg.defaultFromEmail}"
+    "REVERSE_PROXY_TYPE=nginx"
+    "DATABASE_URL=${databaseUrl}"
+    "CACHE_URL=redis://localhost:${toString config.services.redis.port}/0"
+    "MEDIA_ROOT=${cfg.api.mediaRoot}"
+    "STATIC_ROOT=${cfg.api.staticRoot}"
+    "DJANGO_SECRET_KEY=${cfg.api.djangoSecretKey}"
+    "RAVEN_ENABLED=${boolToString cfg.enableRaven}"
+    "RAVEN_DSN=${cfg.ravenDsn}"
+    "MUSIC_DIRECTORY_PATH=${cfg.musicDirectoryPath}"
+    "MUSIC_DIRECTORY_SERVE_PATH=${cfg.musicDirectoryPath}"
+    "FUNKWHALE_FRONTEND_PATH=${cfg.dataDir}/front/dist"
+  ];
+  funkwhaleEnvFileData = builtins.concatStringsSep "\n" funkwhaleEnvironment;
+  funkwhaleEnvScriptData = builtins.concatStringsSep " " funkwhaleEnvironment;
+
+  funkwhaleEnvFile = pkgs.writeText "funkwhale.env" funkwhaleEnvFileData;
+  funkwhaleEnv = {
+    ENV_FILE = "${funkwhaleEnvFile}";
+  };
+in 
+  {
+
+    options = {
+      services.funkwhale = {
+        enable = mkEnableOption "funkwhale";
+
+        user = mkOption {
+          type = types.str;
+          default = "funkwhale";
+          description = "User under which Funkwhale is ran.";
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "funkwhale";
+          description = "Group under which Funkwhale is ran.";
+        };
+
+        database = {
+          host = mkOption {
+            type = types.str;
+            default = "localhost";
+            description = "Database host address.";
+          };
+
+          port = mkOption {
+            type = types.int;
+            default = 5432;
+            defaultText = "5432";
+            description = "Database host port.";
+          };
+
+          name = mkOption {
+            type = types.str;
+            default = "funkwhale";
+            description = "Database name.";
+          };
+
+          user = mkOption {
+            type = types.str;
+            default = "funkwhale";
+            description = "Database user.";
+          };
+
+          password = mkOption {
+            type = types.str;
+            default = "";
+            description = ''
+              The password corresponding to <option>database.user</option>.
+              Warning: this is stored in cleartext in the Nix store!
+              Use <option>database.passwordFile</option> instead.
+            '';
+          };
+
+          passwordFile = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            example = "/run/keys/funkwhale-dbpassword";
+            description = ''
+              A file containing the password corresponding to
+              <option>database.user</option>.
+            '';
+          };
+
+          socket = mkOption {
+            type = types.nullOr types.path;
+            default = "/run/postgresql";
+            defaultText = "/run/postgresql";
+            example = "/run/postgresql";
```

Comment:
No need to specify `defaultText` when the `default` is the same. And no need to specify an example if it's the same as the default